tcs-service/src/test/java/com/transformuk/hee/tis/tcs/service/api/ProgrammeMembershipResourceIntTest.java

```diff
@@ -702,188 +920,256 @@ public void getProgrammeMembershipDto() throws Exception {
     programmeMembershipRepository.saveAndFlush(programmeMembership);
-    ProgrammeMembership updatedProgrammeMembership = programmeMembershipRepository
-        .findById(programmeMembership.getId()).orElse(null);
-    updatedProgrammeMembership.setIntrepidId(UPDATED_INTREPID_ID);
-    updatedProgrammeMembership.setLeavingReason(UPDATED_LEAVING_REASON);
-    ProgrammeMembershipDTO programmeMembershipDTO = programmeMembershipMapper
-        .toDto(updatedProgrammeMembership);
+    programmeMembershipRepository.saveAndFlush(programmeMembership1);
 
-    programmeMembership.setIntrepidId(DEFAULT_INTREPID_ID);
-    programmeMembership.setLeavingReason(DEFAULT_LEAVING_REASON);
-    programmeMembershipRepository.saveAndFlush(programmeMembership);
-    int databaseSizeBeforeUpdate = programmeMembershipRepository.findAll().size();
+    List<ProgrammeMembershipDTO> programmeMembershipDtoList
+       = programmeMembershipMapper.allEntityToDto(Lists.newArrayList(programmeMembership, programmeMembership1));
 
-    //this will update the curriculum membership repository, not the programme membership repository
+    assertThat(programmeMembershipDtoList).hasSize(2); //sanity check
+    ProgrammeMembershipDTO pmDto1 = programmeMembershipDtoList.get(0);
+    pmDto1.setProgrammeMembershipType(UPDATED_PROGRAMME_MEMBERSHIP_TYPE);
+    pmDto1.setProgrammeEndDate(UPDATED_PROGRAMME_END_DATE);
+
+    //when
     restProgrammeMembershipMockMvc.perform(put("/api/programme-memberships")
             .contentType(MediaType.APPLICATION_JSON)
-            .content(TestUtil.convertObjectToJsonBytes(programmeMembershipDTO)))
+            .content(TestUtil.convertObjectToJsonBytes(pmDto1)))
         .andExpect(status().isOk());
 
-    // Validate that the original ProgrammeMembership in the repository is unchanged
-    List<ProgrammeMembership> programmeMembershipList = programmeMembershipRepository.findAll();
-    assertThat(programmeMembershipList).hasSize(databaseSizeBeforeUpdate);
-    ProgrammeMembership testProgrammeMembership = programmeMembershipList
-        .get(programmeMembershipList.size() - 1);
-    assertThat(testProgrammeMembership.getIntrepidId()).isEqualTo(DEFAULT_INTREPID_ID);
-    assertThat(testProgrammeMembership.getLeavingReason()).isEqualTo(DEFAULT_LEAVING_REASON);
+    //then
+    // Validate the updated ProgrammeMembership in the database without a failed 'detached entities' error.
+    ProgrammeMembership retrievedProgrammeMembership = programmeMembershipRepository
+        .findByUuid(programmeMembership.getUuid()).orElse(null);
+    assertThat(retrievedProgrammeMembership.getProgrammeMembershipType()).isEqualTo(UPDATED_PROGRAMME_MEMBERSHIP_TYPE);
+    assertThat(retrievedProgrammeMembership.getProgrammeEndDate()).isEqualTo(UPDATED_PROGRAMME_END_DATE);
   }
 
   @Test
   @Transactional
-  public void updateNonExistingCurriculumMembership() throws Exception {
+  public void updateNonExistingCurriculumMembershipShouldFail() throws Exception {
```

Comment:
I'm unclear whether the existing test still has a place.  If so, the semantics of this aren't the same, i.e. it validates that absence of dates causes an error rather than absence of an id causes an error.

This test should be renamed to identify it is missing start date.